### PR TITLE
Updated entity builder with entity level action eligibility settings

### PR DESCRIPTION
**Pull request by pallavidn**
Added new settings in the entity DTO to define if an entity is eligible for certain actions such as move, provision, suspend.
Server side is available for review. This review updates the turbo-go-sdk with the new go bindings for the protobuf files and also the entity builder.

### CODE REVIEW

File: pkg/builder/entity_dto_builder_test.go [hunk 1]
```diff
@@ -667,6 +676,160 @@ func TestBuildCommodityBoughtFromMap(t *testing.T) {
...
+func TestEntityActionEligibilityFlags(t *testing.T) {
+	trueFlag := true
+	falseFlag := false
+	table := []struct {
+		entityType proto.EntityDTO_EntityType
+		entityId   string
+		suspend    *bool
+		provision  *bool
+	}{
+		// Pods with suspend and proviison flags
```
Comment (dblinn) [inline pkg/builder/entity_dto_builder_test.go]: nit: typo in "provision"

File: pkg/builder/entity_dto_builder.go [hunk 2]
```diff
@@ -345,17 +401,127 @@ func (eb *EntityDTOBuilder) VirtualApplicationData(vAppData *proto.EntityDTO_Vir
...
 			Bought:     commodities,
-		})
+		}
+
+		// update provider type and the action eligibility of the entity across that provider type
+		providerType, providerTypeExists := eb.providerMap[p]
+		if providerTypeExists {
+
+			commodityBought.ProviderType = &providerType
+			aeMap, exists := actionEligibilityMap[providerType]
```
Comment (dblinn) [inline pkg/builder/entity_dto_builder.go]: nit: I don't think "aeMap" is actually a map. It looks like it's a pointer to an ActionEligibilityByProvider struct which makes this code a little bit confusing if you think it's a map.

**Comment by irfanurrehman** [comment]
@pallavidn I reached here looking at https://github.com/turbonomic/kubeturbo/pull/368. I would be interested in reviewing this also, for the sole reason I am interested in improving my understanding of the overall code. But before I could do that there are some things I would really appreciate if you can cleanup the commits in this PR:
- Some recommendations for structuring the commits properly:
1. The actually written code and autogenerated code should be part of different commits. Its makes it super easy for the reviewer to know what to look at.
2. I would really discourage commits like `fmtcheck` and `go.sum`. They simply pollutes commit history. `go.sum` should be part of autogenerate code commit, for example `Update vendored code` and `fmtcheck` is an inherent part of functionality code change.
3. Have meaningful title for a commit. A title like `initial commit for action settings protobuf bindings and builder` has no meaning at all.
- As in one of the commit titles, is `OM-51440`  the epic to look at to understand the changes here?

**Comment by irfanurrehman** [comment]
Apologies, If I am looking at this PR before its ready for review. If so, please label this as `WIP` and let me know once it is.

File: pkg/builder/entity_dto_builder.go (hunk 2 above)
Comment (pallavidn) [reply pkg/builder/entity_dto_builder.go]: Incorporated in new updated PR#81

File: pkg/builder/entity_dto_builder_test.go (hunk 1 above)
Comment (pallavidn) [reply pkg/builder/entity_dto_builder_test.go]: Incorporated in new updated PR#81

**Comment by pallavidn** [comment]
Please refer to PR#81 that contains the newest go bindings and entity builder changes. This is based on the protobuf changes that have been finally committed in the server repository.

**Pull request closed without merging**